### DISCUSSION TITLE
Long connection with MariaDB driver 1.4.0

### DISCUSSION THREAD

**Discussion by infectormp** [dbeaver/dbeaver#327] [Q&A] [answered]
Today i get notification about new version of mariadb driver and update it. But now when i connect to db dbeaver get long tome to connect.

1.4.0
![image](https://cloud.githubusercontent.com/assets/9958813/14177970/57b84ed4-f761-11e5-883b-44cbd40aaf06.png)

1.3.7
![image](https://cloud.githubusercontent.com/assets/9958813/14178023/8ba9f3e6-f761-11e5-9187-145eea949184.png)

**Comment by serge-rider** [comment, marked as answer]
Looks like a problem is in MariaDB drier itself.
Version 1.4.0 returns infinite list of schemas. As a result DBeaver eats all available memory and eventually hangs (I wondering why in your case it still connects - in my case this process never ends).

Just use 1.3.7, I can't fix that. Also you can send a bug report to MariaDB guys.

**Comment by infectormp** [comment]
I suspect that the problem is in the driver. Thank you.
